### DISCUSSION TITLE
XPathAsync deprecated

### DISCUSSION THREAD

**Discussion by trebor678** [hardkoded/puppeteer-sharp#2266] [Q&A]
Now that XPathAsync has been deprecated it suggests using QuerySelectorAsync instead, but when sending it an xpath query like the one below it errors. Do I need to change the syntax?

Example: "//div[@class='test']"

**Comment by kblok** [comment]
@trebor678 you can do `xpath///div[@Class='test']`. `xpath/` is the engine selector and the rest is the xpath itself.

**Reply by Aeroverra** [reply]
I don't quite understand why you would remove this when it could be an extension function that adds that for you. Its much less convienient this way.

**Reply by featherbits** [reply]
yeah, no need to break already running api's

**Reply by kblok** [reply]
The goal of this project is to be a port of puppeteer. If they deprecate, we deprecate.
